### PR TITLE
BFD-1941: Upgrade Out of Date Core Dependencies

### PR DESCRIPTION
**Pull request by JeremyIR**
**JIRA Ticket:**
[BFD-1941](https://jira.cms.gov/browse/BFD-1941)

**User Story or Bug Summary:**

This PR addresses several core library/dependency updates. 

SNYK and Performance Test Results:
- Snyk was run locally, and high and medium priority vulnerabilities no longer show up during scanning
- Performance regression tests were conducted on both this branch with upgraded dependency versions, and the master branch on TEST server. The number of requests differed by 3.60%, and the average number of requests/second differed by 5.42%.

<details>
<summary>
TEST Server Master Branch

</summary>

```
Type     Name                                                                          # reqs      # fails |    Avg     Min     Max    Med |   req/s  failures/s
--------|----------------------------------------------------------------------------|-------|-------------|-------|-------|-------|-------|--------|-----------
GET      /v2/fhir/Coverage search by id                                                   222     0(0.00%) |     38      10     616     14 |    7.10        0.00
GET      /v2/fhir/Coverage search by id / count=10                                        223     0(0.00%) |     44       9     835     14 |    7.13        0.00
GET      /v2/fhir/Coverage search by id / lastUpdated (2 weeks)                           208     0(0.00%) |     31       4     429      6 |    6.65        0.00
GET      /v2/fhir/ExplanationOfBenefit search by id                                       219     0(0.00%) |    439      92    1715    390 |    7.01        0.00
GET      /v2/fhir/ExplanationOfBenefit search by id / count=10                            242     0(0.00%) |    266     108     948    210 |    7.74        0.00
GET      /v2/fhir/ExplanationOfBenefit search by id / lastUpdated / includeTaxNumbers     204     0(0.00%) |    142      89     918    110 |    6.53        0.00
GET      /v2/fhir/Patient search by coverage contract (all pages)                         202     0(0.00%) |     91      27     857     34 |    6.46        0.00
GET      /v2/fhir/Patient search by hashed mbi / includeIdentifiers = mbi                 190     0(0.00%) |     30       4     615      7 |    6.08        0.00
GET      /v2/fhir/Patient search by id                                                    204     0(0.00%) |     29       4     839      6 |    6.53        0.00
GET      /v2/fhir/Patient search by id / _IncludeIdentifiers=mbi / (2 weeks)              205     0(0.00%) |     33       4     664      6 |    6.56        0.00
--------|----------------------------------------------------------------------------|-------|-------------|-------|-------|-------|-------|--------|-----------
Aggregated                                                                      2119     0(0.00%) |    118       4    1715     27 |   67.79        0.00

Response time percentiles (approximated)
Type     Name                                                                                  50%    66%    75%    80%    90%    95%    98%    99%  99.9% 99.99%   100% # reqs
--------|--------------------------------------------------------------------------------|--------|------|------|------|------|------|------|------|------|------|------|------
GET      /v2/fhir/Coverage search by id                                                         14     16     18     22    140    200    220    330    620    620    620    222
GET      /v2/fhir/Coverage search by id / count=10                                              14     16     20     27    150    200    320    380    840    840    840    223
GET      /v2/fhir/Coverage search by id / lastUpdated (2 weeks)                                  6      8     10     11    150    210    260    340    430    430    430    208
GET      /v2/fhir/ExplanationOfBenefit search by id                                            390    470    520    560    700    960   1300   1400   1700   1700   1700    219
GET      /v2/fhir/ExplanationOfBenefit search by id / count=10                                 210    260    300    320    440    660    780    800    950    950    950    242
GET      /v2/fhir/ExplanationOfBenefit search by id / lastUpdated / includeTaxNumbers          110    130    150    170    220    260    380    410    920    920    920    204
GET      /v2/fhir/Patient search by coverage contract (all pages)                               34     43     58    110    210    360    660    660    860    860    860    202
GET      /v2/fhir/Patient search by hashed mbi / includeIdentifiers = mbi                        7      8     10     15     46    170    350    430    620    620    620    190
GET      /v2/fhir/Patient search by id                                                           6      7      8      9     47    180    300    340    840    840    840    204
GET      /v2/fhir/Patient search by id / _IncludeIdentifiers=mbi / (2 weeks)                     6      7     10     12    140    210    380    400    660    660    660    205
--------|--------------------------------------------------------------------------------|--------|------|------|------|------|------|------|------|------|------|------|------
Aggregated                                                                             27    110    180    210    340    490    670    800   1400   1700   1700   2119

```

</details>

<details>
<summary>
TEST Server UPGRADES:

</summary>

```
Type     Name                                                                          # reqs      # fails |    Avg     Min     Max    Med |   req/s  failures/s
--------|----------------------------------------------------------------------------|-------|-------------|-------|-------|-------|-------|--------|-----------
GET      /v2/fhir/Coverage search by id                                                   217     0(0.00%) |     44      10     746     15 |    6.82        0.00
GET      /v2/fhir/Coverage search by id / count=10                                        210     0(0.00%) |     46      10     424     15 |    6.60        0.00
GET      /v2/fhir/Coverage search by id / lastUpdated (2 weeks)                           206     0(0.00%) |     33       4     908      6 |    6.47        0.00
GET      /v2/fhir/ExplanationOfBenefit search by id                                       188     0(0.00%) |    502     123    1881    450 |    5.91        0.00
GET      /v2/fhir/ExplanationOfBenefit search by id / count=10                            203     0(0.00%) |    293      92    1279    250 |    6.38        0.00
GET      /v2/fhir/ExplanationOfBenefit search by id / lastUpdated / includeTaxNumbers     178     0(0.00%) |    171      89    1025    110 |    5.59        0.00
GET      /v2/fhir/Patient search by coverage contract (all pages)                         204     0(0.00%) |     82      26    1139     34 |    6.41        0.00
GET      /v2/fhir/Patient search by hashed mbi / includeIdentifiers = mbi                 207     0(0.00%) |     32       4     907      7 |    6.50        0.00
GET      /v2/fhir/Patient search by id                                                    210     0(0.00%) |     32       4     717      7 |    6.60        0.00
GET      /v2/fhir/Patient search by id / _IncludeIdentifiers=mbi / (2 weeks)              221     0(0.00%) |     50       4     907      7 |    6.94        0.00
--------|----------------------------------------------------------------------------|-------|-------------|-------|-------|-------|-------|--------|-----------
Aggregated                                                                      2044     0(0.00%) |    123       4    1881     21 |   64.21        0.00

Response time percentiles (approximated)
Type     Name                                                                                  50%    66%    75%    80%    90%    95%    98%    99%  99.9% 99.99%   100% # reqs
--------|--------------------------------------------------------------------------------|--------|------|------|------|------|------|------|------|------|------|------|------
GET      /v2/fhir/Coverage search by id                                                         15     17     19     27     67    240    420    480    750    750    750    217
GET      /v2/fhir/Coverage search by id / count=10                                              15     18     22     26    170    250    370    380    420    420    420    210
GET      /v2/fhir/Coverage search by id / lastUpdated (2 weeks)                                  6      7      9     12     83    180    270    370    910    910    910    206
GET      /v2/fhir/ExplanationOfBenefit search by id                                            450    530    590    640    770   1000   1400   1800   1900   1900   1900    188
GET      /v2/fhir/ExplanationOfBenefit search by id / count=10                                 250    290    360    400    490    560    800    910   1300   1300   1300    203
GET      /v2/fhir/ExplanationOfBenefit search by id / lastUpdated / includeTaxNumbers          110    140    170    210    300    490    790    990   1000   1000   1000    178
GET      /v2/fhir/Patient search by coverage contract (all pages)                               34     40     51     60    180    370    430    900   1100   1100   1100    204
GET      /v2/fhir/Patient search by hashed mbi / includeIdentifiers = mbi                        7      8     11     14     46    200    370    380    910    910    910    207
GET      /v2/fhir/Patient search by id                                                           7      8     10     14     95    200    320    380    720    720    720    210
GET      /v2/fhir/Patient search by id / _IncludeIdentifiers=mbi / (2 weeks)                     7      8     12     16    180    260    410    700    910    910    910    221
--------|--------------------------------------------------------------------------------|--------|------|------|------|------|------|------|------|------|------|------|------
Aggregated                                                                             21     98    180    230    380    510    730    910   1800   1900   1900   2044
```

</details>
---

### What Does This PR Do?

Change library versions to more secure , latest value to reduce the vulnerability profile with Snyk.

### What Should Reviewers Watch For?

Does the code follow style guidelines?
Have all the libraries in question been addressed and contribute to lower vulnerability when running snyk?
If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: apps/pom.xml [hunk 1]
```diff
@@ -88,22 +88,22 @@
 		<sonar.jacoco.reportPath>target/jacoco.exec</sonar.jacoco.reportPath>
 		<sonar.jacoco.itReportPath>target/jacoco-it.exec</sonar.jacoco.itReportPath>
 
-		<metrics.version>4.1.29</metrics.version>
-		<newrelic.version>0.5.0</newrelic.version>
+		<metrics.version>4.2.1</metrics.version>
+		<newrelic.version>0.7.0</newrelic.version>
```
Comment (christrotteradhoc) [inline apps/pom.xml:92]: can be combined with below

File: apps/bfd-server/bfd-server-war/pom.xml [hunk 2]
```diff
@@ -205,7 +205,7 @@
 			<!-- Allows us to provide New Relic with additional information on requests. -->
 			<groupId>com.newrelic.agent.java</groupId>
 			<artifactId>newrelic-api</artifactId>
-			<version>5.9.0</version>
+			<version>7.8.0</version>
```
Comment (christrotteradhoc) [inline apps/bfd-server/bfd-server-war/pom.xml:208]: can be gotten from parent

File: apps/pom.xml [hunk 3]
```diff
@@ -88,22 +88,22 @@
...
-		<postgresql.version>42.2.20</postgresql.version>
+		<postgresql.version>42.4.1</postgresql.version>
 		<hsqldb.version>2.6.0</hsqldb.version>
-		<slf4j.version>1.7.7</slf4j.version>
-		<hikari.version>4.0.3</hikari.version>
+		<slf4j.version>1.7.36</slf4j.version>
+		<hikari.version>5.0.1</hikari.version>
 		<flyway.version>8.5.13</flyway.version>
-		<newrelicAgent.version>7.5.0</newrelicAgent.version>
+		<newrelicAgent.version>7.9.0</newrelicAgent.version>
```
Comment (christrotteradhoc) [inline apps/pom.xml:106]: should be renamed to newrelic.agent.version

File: apps/pom.xml [hunk 4]
```diff
@@ -88,22 +88,22 @@
...
-		<aws-java-sdk-bom.version>1.12.245</aws-java-sdk-bom.version>
+		<aws-java-sdk-bom.version>1.12.276</aws-java-sdk-bom.version>
 		<spring-data-jpa.version>2.7.0</spring-data-jpa.version>
-		<hibernate.version>5.6.1.Final</hibernate.version>
+		<hibernate.version>5.6.10.Final</hibernate.version>
 		<junit.version>5.8.2</junit.version>
 		<jackson.version>2.13.3</jackson.version>
 		<mockito.version>4.2.0</mockito.version>
-		<postgresql.version>42.2.20</postgresql.version>
+		<postgresql.version>42.4.1</postgresql.version>
```
Comment (christrotteradhoc) [inline apps/pom.xml:101]: please take out the old postgres.driver.version

File: apps/pom.xml [hunk 5]
```diff
@@ -233,10 +233,12 @@
 					Hibernate is used for that in our projects.) -->
 				<artifactId>hibernate-jpa-2.1-api</artifactId>
 				<groupId>org.hibernate.javax.persistence</groupId>
-				<version>1.0.0.Final</version>
+				<version>1.0.2.Final</version>
 			</dependency>
+
+			<!-- Hibernate version not latest because entity manager does not have a 6.1 to upgrade to-->
```
Comment (christrotteradhoc) [inline apps/pom.xml:239]: I think we can get rid of this comment in favor of one on the version itself

File: apps/pom.xml [hunk 6]
```diff
@@ -88,22 +88,22 @@
...
-		<newrelic.version>0.5.0</newrelic.version>
+		<metrics.version>4.2.1</metrics.version>
+		<newrelic.version>0.7.0</newrelic.version>
 		<dropwizard-metrics-newrelic.version>0.7.0</dropwizard-metrics-newrelic.version>
 		<telemetry-http-okhttp.version>0.15.0</telemetry-http-okhttp.version>
-		<aws-java-sdk-bom.version>1.12.245</aws-java-sdk-bom.version>
+		<aws-java-sdk-bom.version>1.12.276</aws-java-sdk-bom.version>
 		<spring-data-jpa.version>2.7.0</spring-data-jpa.version>
-		<hibernate.version>5.6.1.Final</hibernate.version>
+		<hibernate.version>5.6.10.Final</hibernate.version>
```
Comment (christrotteradhoc) [inline apps/pom.xml:96]: we should have a comment here along these lines:

```
Newer (6.0) release versions for hibernate are available for some hibernate components but not all (in particular, entity-manager does not have a 6.0 version available yet). Staying on latest 5.x version until all components have 6.0 versions available.
```

File: apps/pom.xml (hunk 4 above)
Comment (JeremyIR) [reply apps/pom.xml:100]: Removed

File: apps/pom.xml [hunk 7]
```diff
@@ -264,6 +264,8 @@
 				<artifactId>activation</artifactId>
 				<version>1.1.1</version>
 			</dependency>
+
+			<!-- Dependencies are in the Jakarta namespace. Should be upgraded when Jetty is upgraded to version 11 -->
```
Comment (christrotteradhoc) [inline apps/pom.xml:268]: New comment suggestion:

```
BFD currently uses Jetty 10 which is compatible with Java EE 8 and predates the move to the jakarta namespace
that comes with Jakarta EE 9. For the sake of consistency, the other EE APIs and implementations are also being held at their Java EE 8 compatible, pre-jakarta namespaced versions.

Once BFD moves to Jetty 11 (which is dependent on the HAPI dependency becoming compliant with Jakarta EE
9), these dependencies below should be updated as well which will necessitate moving all BFD javax imports to the
jakarta versions.
```

File: apps/pom.xml (hunk 3 above)
Comment (JeremyIR) [reply apps/pom.xml]: Updated

File: apps/bfd-server/bfd-server-war/pom.xml (hunk 2 above)
Comment (JeremyIR) [reply apps/bfd-server/bfd-server-war/pom.xml]: Updated

File: apps/pom.xml (hunk 1 above)
Comment (JeremyIR) [reply apps/pom.xml]: Updated

File: apps/pom.xml (hunk 5 above)
Comment (JeremyIR) [reply apps/pom.xml:237]: Updated

File: apps/pom.xml (hunk 7 above)
Comment (JeremyIR) [reply apps/pom.xml]: Updated

File: apps/pom.xml [hunk 8]
```diff
@@ -264,6 +266,12 @@
 				<artifactId>activation</artifactId>
 				<version>1.1.1</version>
 			</dependency>
+
+			<!-- BFD currently uses Jetty 10 which is compatible with Java EE 8
+			and predates the move to the jakarta namespace that comes with Jakarta EE 9.
+			For the sake of consistency, the other EE APIs and implementations are also being held at their Java EE 8 compatible, pre-jakarta namespaced versions.
+			Once BFD moves to Jetty 11 (which is dependent on the HAPI dependency becoming compliant with Jakarta EE 9),
+			these dependencies below should be updated as well which will necessitate moving all BFD javax imports to the jakarta versions. -->
```
Comment (christrotteradhoc) [inline apps/pom.xml:274]: minor nit -- would you mind cleaning up the formatting here to have more consistent line breaks?

**Review by christrotteradhoc — APPROVED** [review]

**Review by mackec1 — APPROVED** [review]
Looks reasonable.

**Pull request merged**